src/order/complete_well_founded.lean

```diff
@@ -49,6 +49,65 @@ same `Sup`. -/
 def is_Sup_finite_compact : Prop :=
 ∀ (s : set α), ∃ (t : finset α), ↑t ⊆ s ∧ Sup s = t.sup id
 
+/-- An element k of a complete lattice is said to be compact if any set that Sups
+above k has a finite subset that Sups above k -/
+def is_compact_element {α : Type*} [complete_lattice α] (k : α) :=
+∀ s : set α, k ≤ Sup s → ∃ t : finset α, ↑t ⊆ s ∧ k ≤ t.sup id
+
+/-- An element k of a lattice is said to be finite if any directed
+set that Sups above k already got above k at some point in the set. -/
+def is_finite_element {α : Type*} [complete_lattice α] (k : α) :=
```

Comment:
I know this is going contrary to @ocfnash 's advice, but I don't think there should be separate names for `is_compact_element` and `is_finite_element`. At least, if both are only defined for elements of a `complete_lattice`, where they are equivalent. If there's a natural way to define these predicates outside the `complete_lattice` context, where they're not equivalent, then do it, but otherwise, I think we're just creating two names for the same thing, which complicates the API.

Reply:
I still lean in favour of the two names but I'm happy to defer to @awainverse on this point!

Reply:
The source that [this page](https://encyclopediaofmath.org/wiki/Compact_lattice_element) cites defines them both for complete lattices, but remarks that the "finite" defn. makes sense for lattices where only Sups of directed sets necessarily exist. But, it's mostly a moot point and I think the only reason there are two definitions is that "finite" came from computer science.

If we don't use both names, then how would you recommend naming the iff lemma? Something like the following?
```
theorem is_compact_element_iff_le_of_directed_Sup_le (k : α) :
  is_compact_element k ↔ ∀ s : set α, s.nonempty → directed_on (≤) s → k ≤ Sup s → ∃ x : α, x ∈ s ∧ k ≤ x
```

Reply:
That seems fine to me.

Reply:
You could also maybe add a docstring saying that this property also defines finite elements, so that if someone does have a use for that idea separately, they know where to look.